q2_stats/_stats.py

```diff
@@ -231,9 +236,25 @@ def _compare_wilcoxon(group_a, group_b, alternative, p_val_approx) -> dict:
         'n': len(filtered.index),
     }
 
-    stat, p_val = scipy.stats.wilcoxon(
-        filtered.iloc[:, 0], filtered.iloc[:, 1],
-        nan_policy='raise', mode=p_val_approx, alternative=alternative)
+    if filtered.empty:
+        if ignore_empty_comparator:
+            stat = float('nan')
+            p_val = float('nan')
+        elif not ignore_empty_comparator:
```

Comment:
For a boolean, there's only two possible branches, so the following is more typical:

```suggestion
        else:
```

(otherwise the reader might have to decide if there's something they're missing to warrant the `elif`)